contracts/math/BondingCurve.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/IERC721EnumerableUpgradeable.sol";
+
+contract BondingCurve {
+    uint256 constant one = 10**18;
```

Comment:
im not sure if this contract will remain inheritable or if it will become a library so i don't want to contextualize the token limit but i did abstract this into a constant